scripts/compile.py

```diff
@@ -0,0 +1,164 @@
+    url_pattern = url_pattern.replace(r"^https?:\/\/(?:[a-z0-9-]+\.)*?", "", 1)
+    url_pattern = url_pattern.replace(r"https?:\/\/([a-z0-9-.]*\.)", "", 1)
+    url_pattern = url_pattern.replace(r"^https?:\/\/", "", 1)
+    # domain= style TLD globbing
+    url_pattern = url_pattern.replace(r"(?:\.[a-z]{2,}){1,}", ".*", 1)
+    # Remove backslashes
+    url_pattern = url_pattern.replace("\\", "")
+
+    # Specific fixups
+    url_pattern = url_pattern.replace("(?:accounts.)?", "", 1)
+    url_pattern = url_pattern.replace("(?:support.)?", "", 1)
+    url_pattern = url_pattern.replace("(?:yandex.*|ya.ru)", "yandex.*", 1)
+
+    return url_pattern
+
+
+def normalize_exception(exception: str) -> str:
+    orig_exception = exception
+
+    exception = exception.replace(r"^https?:\/\/(?:[a-z0-9-]+\.)*?", "||", 1)
+    exception = exception.replace(r"^https?:\/\/", "||", 1)
+    # FIXME: |ws://
+    exception = exception.replace(r"^wss?:\/\/(?:[a-z0-9-]+\.)*?", "|wss://", 1)
+    exception = exception.replace(r"(?:\.[a-z]{2,}){1,}", "TLD_WILDCARD", 1)
+
+    exception = exception.replace("=[^/?&]*", "=")
+    exception = exception.replace("=.*?", "=")
+    exception = exception.replace("=.", "=")
+    exception = exception.replace("[^?]*\\?.*?", "*?*")
+    exception = exception.replace("[^?]+.*?&?", "*?*")
+    exception = exception.replace("\\?.*?", "?")
+    exception = exception.replace(".*?&?", "*")
+    exception = exception.replace(".*?", "*")
+
+    exception = exception.replace("\\", "")
+
+    if any(c in "([" for c in exception):
+        exception = orig_exception
+        exception = exception.replace("(?:", "(")
+        return "regex", exception
+    elif any(c in "/?" for c in exception):
+        exception = exception.replace("TLD_WILDCARD", ".*", 1)
+        exception = exception.replace("|wss://zoom.us", "|wss://zoom.us^", 1)
+        return "path", exception
+    else:
+        exception = exception.replace("TLD_WILDCARD", ".*", 1)
+        exception = exception.replace("||", "", 1)
+        return "domain", exception
+
+
+def expand_se(rule: str) -> [str]:
+    # https://stackoverflow.com/questions/20061268/python-regex-string-expansion
+    # Is there a lib for that?
+    #
+    # 1. foo_(1|2)_bar -> foo_1_bar + foo_2_bar
+    # 2. foo_[12]_bar -> foo_1_bar + foo_2_bar
+    # 3. foo_?bar -> foobar + foo_bar
+    # But foo_[a-z]*_bar -> foo_[a-z]*_bar
+    raise NotImplementedError
+
+
+def is_regex(rule: str) -> bool:
+    return any(c in r".^$*+?{}[]\|()" for c in rule)
+
+
+def print_rules(
+    url_pattern: str, rules: [str], regex_fromat: str, plain_format: str
+) -> None:
+    for rule in rules:
+        if is_regex(rule):
+            endrules.write(regex_fromat.format(rule, url_pattern) + "\n")
+        else:
+            endrules.write(plain_format.format(rule, url_pattern) + "\n")
+
+def getrules():
+  import requests
+  return requests.get("https://raw.githubusercontent.com/ClearURLs/Rules/master/data.min.json").text
```

Comment:
```suggestion
def getrules() -> str:
  RULES = "https://raw.githubusercontent.com/ClearURLs/Rules/master/data.min.json"
  return requests.get(RULES).text


```

Reply:
Ah. I'm not great at code styling, but I did once read about something once, so I will push this